packages/babel/types/index.d.ts

```diff
@@ -0,0 +1,102 @@
+import { Plugin } from 'rollup';
+import { FilterPattern } from '@rollup/pluginutils';
+import * as babelCore from '@babel/core';
+
+export interface RollupBabelInputPluginOptions
+  extends Omit<babelCore.TransformOptions, 'include' | 'exclude'> {
+  /**
+   * A minimatch pattern, or array of patterns, which specifies the files in the build the plugin should operate on. When relying on Babel configuration files you cannot include files already excluded there.
+   * @default undefined;
+   */
+  include?: FilterPattern;
+  /**
+   * A minimatch pattern, or array of patterns, which specifies the files in the build the plugin should ignore. When relaying on Babel configuration files you can only exclude additional files with this option, you cannot override what you have configured for Babel itself.
+   * @default undefined;
+   */
+  exclude?: FilterPattern;
+  /**
+   * An array of file extensions that Babel should transpile. If you want to transpile TypeScript files with this plugin it's essential to include .ts and .tsx in this option.
+   * @default ['.js', '.jsx', '.es6', '.es', '.mjs']
+   */
+  extensions?: string[];
+  /**
+   * It is recommended to configure this option explicitly (even if with its default value) so an informed decision is taken on how those babel helpers are inserted into the code.
+   * @default 'bundled'
+   */
+  babelHelpers?: 'bundled' | 'runtime' | 'inline' | 'extends';
+  /**
+   * Before transpiling your input files this plugin also transpile a short piece of code for each input file. This is used to validate some misconfiguration errors, but for sufficiently big projects it can slow your build times so if you are confident about your configuration then you might disable those checks with this option.
+   * @default false
+   */
+  skipPreflightCheck?: boolean;
+}
+
+export interface RollupBabelOutputPluginOptions
+  extends Omit<babelCore.TransformOptions, 'include' | 'exclude'> {
+  /**
+   * Use with other formats than UMD/IIFE.
+   * @default false
+   */
+  allowAllFormats?: boolean;
+}
+
+export type CustomInputPluginOptions = (
+  options: RollupBabelInputPluginOptions & Record<string, any>
+) => {
+  customOptions: Record<string, any>;
+  pluginOptions: RollupBabelInputPluginOptions;
+};
+export type CustomOutputPluginOptions = (
+  options: RollupBabelOutputPluginOptions & Record<string, any>
+) => {
+  customOptions: Record<string, any>;
+  pluginOptions: RollupBabelOutputPluginOptions;
+};
+export type CustomPluginConfig = (
```

Comment:
Hm, I find it a little bit odd that types related to the most common exports have very specific names like `RollupBabelInputPluginOptions` but those related to more esoteric exports have rather generic names like `CustomPluginConfig`. I believe that we should unify those - one way or another.

I understand those specific names are getting really mouthful though - what is the convention used by other plugins in this monorepo? maybe exporting a `RollupBabel` namespace would make this slightly more approachable? with that we could have types like `RollupBabel.InputPluginOptions` etc

Reply:
👍  Done